### PR TITLE
MM-30306 - CRT beta notice

### PR DESCRIPTION
**Pull request by esethna**
Let's please merge this before the v5.37 release @amyblais

### CODE REVIEW

**Comment by esethna** [comment]
Can we please test the markdown, line breaks, server config condition and also make sure that clicking the link to docs doesn't close the modal?

File: notices.json [hunk 1]
```diff
@@ -72,5 +72,24 @@
...
+    "conditions": {
+      "audience": "all",
+      "clientType": "all",
+      "serverVersion": [">5.36"],
+      "displayDate": ">= 2021-07-19T00:00:00Z",
+      "serverConfig": { "ServiceSettings.CollapsedThreads": "default-off" }
+    },
+    "localizedMessages": {
+      "en": {
+        "title": "Collapsed Reply Threads are here, get early access to the beta",
```
Comment (cwarnermm) [inline notices.json:87]:
```suggestion
        "title": "Collapsed Reply Threads are here - get early access to the beta",
```

File: notices.json [hunk 2]
```diff
@@ -72,5 +72,24 @@
...
+        "description": "You can now enable an early beta of Collapsed Reply Threads in **Account Settings** > **Display** > **Collapsed Reply Threads (Beta)**\n\nPlease [review the list of known issues](https://docs.mattermost.com/help/messaging/organizing-conversations.html) as we work towards stabilizing the feature.",
```
Comment (cwarnermm) [inline notices.json:88]:
```suggestion
        "description": "You can now enable an early beta of Collapsed Reply Threads in **Account Settings** > **Display** > **Collapsed Reply Threads (Beta)**\n\nPlease [review the list of known issues](https://docs.mattermost.com/help/messaging/organizing-conversations.html) as we work towards stabilizing this feature.",
```

**Review by cwarnermm — DISMISSED** [review]
Non-blocking suggestions provided inline

**Comment by esethna** [comment]
I just added a second notice here that will appear for admins to guide them to enable CRT

File: notices.json [hunk 3]
```diff
@@ -72,5 +72,42 @@
...
     }
+  },
+  {
+    "id": "collapsed-reply-threads-beta-user",
+    "conditions": {
+      "audience": "all",
+      "clientType": "all",
+      "serverVersion": [">5.36"],
+      "displayDate": ">= 2021-07-19T00:00:00Z",
+      "serverConfig": { "ServiceSettings.CollapsedThreads": "default-off" }
```
Comment (esethna) [inline notices.json:83]: @ogi-m can we confirm that this config checks works on both cloud and on-prem?

**Comment by ogi-m** [comment]
@esethna I've attached the screenshots of the notices.
The config condition value needs to be changed from "default-off" to "default_off".
Verified the condition works for both cloud and onprem, and that the modal isn't dismissed when clicking on the link.

A few questions:
1. Should we be adding a config check to the admin notice so that admins that have already enabled CRT on the server don't receive the notice?
2. Do we want to add a user config check to the user notice so that user's that have already enabled CRT in account settings don't receive the notice?
2. Do we want to show the user notice to admins as well, since they already get the notice telling them to enable CRT, and then when they enable it they get a similar notice, except it now tells them to enable it in account settings?

SharedScreenshot123
SharedScreenshot1234

**Comment by esethna** [comment]
Thanks @ogi-m 

1. Ya good point, feel free to add the condition for "serverConfig": { "ServiceSettings.CollapsedThreads": "off" }
2. Sure, if user config conditions are working and tested let's do that
3. Another good point, we can scope the user notice to just end users (exclude admins)

Feel free to edit as needed, appreciate the thorough review!

**Comment by esethna** [comment]
@ogi-m can you also file a bug for the styling of this modal, it's horrible:

spacing, text color, alignment etc all need tweaking

image

**Comment by ogi-m** [comment]
@esethna Clicking the link doesn't dismiss the notice, but clicking on "Learn more" does

**Comment by ogi-m** [comment]
1. Done
2. Can't figure out the correct config name/value at the moment, I've asked about it in our channel.
3. Looks like we can't actually do that. We only have the option for targeting _all_ or _sysadmins_, but not members.

@esethna Opened a ticket for the styling issue [here](https://mattermost.atlassian.net/browse/MM-37189). I've referenced the figma designs, but please take a look and add any necessary details.

**Review by ogi-m — APPROVED** [review]
@esethna I've updated the known issues URL so that it links directly to that section, and I've shortened the id's so that the notices don't repeat.

**Comment by amyblais** [comment]
Eric I wanted to let you check and confirm that this is good to go before I merge this.

**Comment by esethna** [comment]
@amyblais @ogi-m probably we should off on merging until this is investigated and resolved: https://community-daily.mattermost.com/private-core/pl/prhg6cyiupbrtyas64jiz3taqe

**Comment by ogi-m** [comment]
@esethna if you're referring to the repeating notice, that's been resolved

**Comment by esethna** [comment]
Ah okay great @ogi-m , then I think we're good to merge this @amyblais

**Pull request merged**